src/hooks/useActiveDescendantListFocus/__tests__/useActiveDescendantListFocus.jest.js

```diff
@@ -78,37 +141,144 @@ function runListUnitTest(isHorizontal) {
       userEvent.keyboard(moveForwardKey);
     });
 
-    expect(result.current.visualFocusItemIndex).toEqual(1);
+    expect(result.current.visualFocusItemIndex).toEqual(3);
+  });
+
+  it("should not return any visual focus if wrapper is not focused", async () => {
+    const onItemClick = jest.fn();
+    const { result } = renderHookForTest({ onItemClick, isHorizontal, defaultVisualFocusFirstIndex });
+
+    expect(result.current.visualFocusItemIndex).toEqual(undefined);
   });
 
   it("should not navigate to next item when user try to navigate by using keys for the  opposite dimension to the list dimension ", async () => {
     const onItemClick = jest.fn();
-    const { result } = renderHookForTest({ onItemClick, isHorizontal });
+    const { result } = renderHookForTest({ onItemClick, isHorizontal, defaultVisualFocusFirstIndex });
 
     act(() => {
       // set focus on the list's element which in charge on natural focus element
       element.focus();
+      // move visual focus to first item
+      userEvent.keyboard(oppositeMoveForwardKey);
+    });
 
+    let before = result.current.visualFocusItemIndex;
+    act(() => {
       // move visual focus to first item
       userEvent.keyboard(oppositeMoveForwardKey);
     });
 
-    expect(result.current.visualFocusItemIndex === 0).toBeFalsy();
+    expect(result.current.visualFocusItemIndex).toEqual(before);
+  });
+  it("should change visually focused item when calling setVisualFocusItemId with triggeredWithKeyboard argument", async () => {
+    const { result } = renderHookForTest({ isHorizontal, defaultVisualFocusFirstIndex });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+
+    act(() => {
+      const isTriggeredByKeyboard = true;
+      // Change visually focused item by using setVisualFocusItemId function
+      result.current.setVisualFocusItemId(THIRD_ITEM_ID, isTriggeredByKeyboard);
+    });
+
+    // Now the visual focus item index should be the index of THIRD_ITEM_ID item
+    expect(result.current.visualFocusItemIndex).toEqual(3);
   });
 }
 
+const combineFeatures = features => {
```

Comment:
I agree :) as we discuss will do in different pr

Reply:
I feel it will not be more clearer with the new name... it only will be a longer name.
and we say that we want to refactor these tests in the near future... so I prefer to keep this like this for if it's not necessary